### PR TITLE
[Merged by Bors] - Mention world_query(ignore) attribute for WorldQuery derivation

### LINKED ISSUE #6283: `WorldQuery` docs do not mention `#[world_query(ignore)]` behaviour
[`WorldQuery`](http://dev-docs.bevyengine.org/bevy/ecs/query/trait.WorldQuery.html) docs mention how to use the `derive(WorldQuery)` macro but don't mention the `#[world_query(ignore)]` functionality. [`SystemParam`](http://dev-docs.bevyengine.org/bevy/ecs/system/trait.SystemParam.html#attributes) does mention its `#[system_param(ignore)]` functionality.

### PR DESCRIPTION
**Pull request by x3ro**
# Objective

Add documentation `#[world_query(ignore)]`. Fixes #6283.

---

I've only described it's behavior so far (which appears to be the same as with `system_param`). Is there another use-case for this besides with `PhantomData`? I could only find a single usage of this construct on GitHub, which is [here](https://github.com/tqwewe/bevy-editor-2/blob/ffcb816927a1bbdcf1cb0136ce47864e5040f9fb/bevy/examples/ecs/custom_query_param.rs#L102). 

I was also wondering if it would make sense to add a usage example to the `custom_query_example`? 🤔 That's why it's currently still in there.

### CODE REVIEW

File: crates/bevy_ecs/src/query/fetch.rs [hunk 1]
```diff
@@ -55,7 +55,12 @@ use std::{cell::UnsafeCell, marker::PhantomData};
 /// - Methods can be implemented for the query items.
 /// - There is no hardcoded limit on the number of elements.
 ///
-/// This trait can only be derived if each field of the struct also implements `WorldQuery`.
+/// This trait can only be derived if each field either
+///
+/// * also implements `WorldQuery`, or
+/// * is marked with `#[world_query(ignore)]`. Fields decorated with this attribute
+///   will be created with the default value upon realisation.
```
Comment (james7132) [inline crates/bevy_ecs/src/query/fetch.rs:62]: Please explicitly mention that `Default` is both required to be implemented and used in this case.

Comment (x3ro) [reply crates/bevy_ecs/src/query/fetch.rs]: Not 100% sure what you mean by `Default` is [...] required to be [...] used`. Can you elaborate? For now I've only added that it needs to be implemented for the type.

**Comment by x3ro** [comment]
Hey :) Do you think the PR can be merged like this, or should I make additional changes?

File: crates/bevy_ecs/src/query/fetch.rs [hunk 2]
```diff
@@ -55,7 +55,13 @@ use std::{cell::UnsafeCell, marker::PhantomData};
 /// - Methods can be implemented for the query items.
 /// - There is no hardcoded limit on the number of elements.
 ///
-/// This trait can only be derived if each field of the struct also implements `WorldQuery`.
+/// This trait can only be derived if each field either
+///
+/// * also implements `WorldQuery`, or
+/// * is marked with `#[world_query(ignore)]`. Fields decorated with this attribute
+///   will be created with the default value upon realisation, so `Default` has to
+///   be implemented on its type.
```
Comment (james7132) [inline crates/bevy_ecs/src/query/fetch.rs:63]:
```suggestion
/// * is marked with `#[world_query(ignore)]`. Fields decorated with this attribute
///   must implement [`Default`] and will be initialized to the default value as defined
///   by the trait.
```

**Review by james7132 — CHANGES_REQUESTED** [review]
LGTM otherwise.

**Review by james7132 — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**